File: Lua/d2.lua
s=... -- input
-- Part 1
t=tonumber c=0 p=("").gmatch for x in p(s,"[^\n]+")do f=p(x,"%d+")l,w,h=t(f()),t(f()),t(f())y={l*w,w*h,h*l}table.sort(y)c=c+3*y[1]+2*(y[2]+y[3])end print(c)
-- Part 2
t=tonumber c=0 p=("").gmatch for x in p(s,"[^\n]+")do f=p(x,"%d+")l,w,h=t(f()),t(f()),t(f())y={l,w,h}table.sort(y)c=c+2*(y[1]+y[2])+l*w*h end print(c)
-- Both parts
t=tonumber c=0 d=0 b=table.sort p=("").gmatch for x in p(s,"[^\n]+")do f=p(x,"%d+")l,w,h=t(f()),t(f()),t(f())y={l*w,w*h,h*l}b(y)c=c+3*y[1]+2*(y[2]+y[3])y={l,w,h}b(y)d=d+2*(y[1]+y[2])+l*w*h end p=print p(c)p(d)

File: C/d2.c
#include <stdio.h>
int main(){
    const char*s=""; /* input */
    /* Part 1 */ {
    const char*f=s;
    int l,w,h,x,y,z,n,c=0;
    while(sscanf(f,"%dx%dx%d\n%n",&l,&w,&h,&n)>2){
        f+=n;
        x=l*w;
        y=w*h;
        z=h*l;
        c+=2*(x+y+z)+(x<y&&x<z?x:y<z?y:z);
    }
    printf("%d\n",c);
    } /* Part 2 */ {
    const char*f=s;
    int l,w,h,n,c=0;
    while(sscanf(f,"%dx%dx%d\n%n",&l,&w,&h,&n)>2){
        f+=n;
        c+=2*(l<w?(w<h?l+w:l+h):l<h?w+l:w+h)+l*w*h;
    }
    printf("%d\n",c);
    } /* Both parts */ {
    const char *f=s;
    int l,w,h,x,y,z,n,c=0,d=0;
    while(sscanf(f,"%dx%dx%d\n%n",&l,&w,&h,&n)>2){
        f+=n;
        x=l*w;
        y=w*h;
        z=h*l;
        c+=2*(x+y+z)+(x<y&&x<z?x:y<z?y:z);
        d+=2*(l<w?(w<h?l+w:l+h):l<h?w+l:w+h)+l*w*h;
    }
    printf("%d\n%d\n",c,d);
}}
